src/test/operators/table_scan_test.cpp

```diff
@@ -283,4 +302,101 @@ TEST_F(OperatorsTableScanTest, ScanOnReferenceSegmentWithNullValue) {
   }
 }
 
+TEST_F(OperatorsTableScanTest, ScanWithNotComparableSearchValue) {
+  auto scan_1 = std::make_shared<TableScan>(_table_wrapper, ColumnID{0}, ScanType::OpGreaterThanEquals, "Hello");
+  EXPECT_THROW(scan_1->execute(), std::logic_error);
+
+  auto scan_2 = std::make_shared<TableScan>(_table_wrapper, ColumnID{0}, ScanType::OpGreaterThanEquals, 123.5);
+  EXPECT_THROW(scan_2->execute(), std::logic_error);
+
+  auto scan_3 = std::make_shared<TableScan>(_table_wrapper, ColumnID{0}, ScanType::OpGreaterThanEquals, 123.5f);
+  EXPECT_THROW(scan_3->execute(), std::logic_error);
+}
+
+TEST_F(OperatorsTableScanTest, GetValuesOfProtectedVariables) {
+  auto scan = std::make_shared<TableScan>(_table_wrapper, ColumnID{0}, ScanType::OpGreaterThanEquals, 124);
+
+  EXPECT_EQ(scan->column_id(), ColumnID{0});
+  EXPECT_EQ(scan->search_value(), AllTypeVariant{124});
+  EXPECT_EQ(scan->scan_type(), ScanType::OpGreaterThanEquals);
+}
+
+TEST_F(OperatorsTableScanTest, EmptyChunkIfFilterConditionNotMet) {
+  auto scan = std::make_shared<TableScan>(_table_wrapper, ColumnID{0}, ScanType::OpEquals, 1);
+  scan->execute();
+  EXPECT_EQ(scan->get_output()->get_chunk(ChunkID{0})->size(), 0);
+}
+
+TEST_F(OperatorsTableScanTest, SegmentsReferencingTheSameTableHaveEqualPositionLists) {
+  const auto scan = std::make_shared<TableScan>(_table_wrapper, ColumnID{0}, ScanType::OpGreaterThanEquals, 124);
+  scan->execute();
+  const auto output = scan->get_output();
+  const auto referenceSegment0 =
```

Comment:
camelCase when everything else is snake_case
![](https://media.giphy.com/media/VZBQzXBEtZTdTGkQKT/giphy.gif)